brats/data/datasets.py

```diff
@@ -21,14 +22,31 @@ def __init__(self, root: str, transform=None):
         self._files = list(os.scandir(root))
         self._transform = transform
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._files)
 
-    def __getitem__(self, item):
-        scan = nib.load(self._files[item].path)
+    def __getitem__(self, idx: int) -> typing.Any:
+        scan = nib.load(self._files[idx].path)
         scan_array = scan.get_fdata()
 
         if self._transform:
             scan_array = self._transform(scan_array)
 
         return scan_array
+
+
+class CombinedDataset(data.Dataset):
+    """
+    Takes multiple datasets of the same length and combines them.
+    On `__getitem__(n)` it returns a tuple containing nth element of each dataset.
+    """
+
+    def __init__(self, *datasets: data.Dataset):
+        assert all(len(dataset) == len(datasets[0]) for dataset in datasets), "Length of both datasets must match"
```

Comment:
Just a neatpick `all(len(dataset) == len(datasets[0]) for dataset in datasets[1:])`. Also
"Length of **all** datasets must be the same".

Reply:
Yeah, good catch 😄